src/olympia/users/tests/test_tasks.py

```diff
@@ -172,3 +178,138 @@ def test_resize_photo_poorly():
     # assert nothing happened
     src_image = Image.open(src.name)
     assert src_image.size == (339, 128)
+
+
+def list_to_csv(data):
+    output = io.StringIO()
+    writer = csv.writer(output)
+    writer.writerow(['Zero', 'One', 'Two', 'Email'])
+
+    for item in data:
+        row = ['0', '1', '2', item if item is not None else '']
+        writer.writerow(row)
+    return output.getvalue()
+
+class TestEmailBlock(TestCase):
+    def test_retry_if_api_returns_bad_response(self):
+        responses.add(
+            responses.GET,
+            f"{settings.SOCKET_LABS_HOST}/servers/{settings.SOCKET_LABS_SERVER_ID}/suppressions/download?sortField=suppressionLastUpdate&sortDirection=dsc",
+            status=500,
+        )
+
+        with raises(HTTPError):
+            sync_blocked_emails()
+
+        # Asser that task is retried
```

Comment:
Have not figured out how to implement this test, expecting a retry.. It might actually be that the task is not retrying... I'm not 100% sure.

I tried replaceing line 201 with

```python
with raises(Retry):
```

And this test failed claiming the exception raised was an HTTPError, but that is the exception that should trigger the retry.. so not totally sure what to do here.